pkg/clients/sns/snssubscription.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sns
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sns"
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/crossplane/provider-aws/apis/notification/v1alpha1"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+const (
+	//SNSSubscriptionNotFound is the error code that is returned if SNS Subscription is not present
+	SNSSubscriptionNotFound = "InvalidSNSSubscription.NotFound"
+
+	// SNSSubscriptionInPendingConfirmation will be raise when SNS Subscription is found
+	// but in "pending confirmation" state.
+	SNSSubscriptionInPendingConfirmation = "InvalidSNSSubscription.PendingConfirmation"
+)
+
+// IsErrorSubscriptionNotFound returns true if the error code indicates that the item was not found
+func IsErrorSubscriptionNotFound(err error) bool {
```

Comment:
Nit: I don't think we usually include `Error` in these kind of checks - so this would just be `IsSubscriptionNotFound`. Ditto for other similar checks in this package.